### ISSUE TITLE
Maven resource filtering not fully supported in Spring Boot

### ISSUE THREAD

**Issue by andreasfranz** [spring-projects/spring-boot#4992]
As described in the [Spring Boot Documentation](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-properties-and-configuration.html#howto-use-short-command-line-arguments) you must use i.e. @maven.token@ instead of ${maven.token}. For the simple case @pom.version@ this works for me but I tried to get the same result as ${pom.version:major} by using @pom.version:major@ which does not work. @pom.version@ get repleaced by e.g. 1.0.0 but @pom.version:major@ stays as it is.

**Comment by wilkinsona** [comment]
Spring Boot doesn't do anything special here – it simply configures the resources plugin to use a custom delimiter – so I suspect that you've found a Maven bug. What version are you using?

**Comment by andreasfranz** [comment]
I use spring-boot-starter-parent 1.3.1.RELEASE, built my project with the spring-boot-maven-plugin. I also included the Spring platform-bom 2.0.1.RELEASE. The current maven version is the latest 3.3.9

**Comment by snicoll** [comment]
@andreasfranz I think that you are confusing two completely different features. Spring placeholders (`${}` does support some features such as default value (`${hello.msg:World}`). Maven placeholders don't as far as I know. 

Before 1.3, you could use both of them with the same syntax which is super confusing so we decided to go for an alternative (`@@`). It looks like you're trying to use a Spring placeholder feature with a Maven placeholder.

The best explanation I can give you regarding  `${pom.version:major}` is the following:

* If `pom.version` is set then it's filtered by Maven (it's always there so I don't even get the default)
* If `pom.version` is not there, then it's not filtered out and Spring does process it (either replacing the value from the `Environment` or applying the default).

Long story short, you can't use default values with Maven placeholders. Does that make sense?

**Comment by snicoll** [comment]
Thanks for updating your original description. That confirms what I just explained above. This is also referenced in the [release notes](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-1.3-Release-Notes#maven-resources-filtering)
